File: EmployeeManagementAPI/Models/Employee.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace EmployeeManagementAPI.Models
{
    public class Employee : Person
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int EmployeeId { get; set; }
        [Required]
        [StringLength(128)]
        [Display(Name = "Employee Number")]
        public string EmployeeNum { get; set; }

        [Required]
        [DataType(DataType.Date)]
        [DisplayFormat(DataFormatString = "{0:yyyy-MM-dd}", ApplyFormatInEditMode = true)]
        [Display(Name = "Employee Date")]
        public DateTime EmployedDate { get; set; }

        [DataType(DataType.Date)]
        [DisplayFormat(DataFormatString = "{0:yyyy-MM-dd}", ApplyFormatInEditMode = true)]
        [Display(Name = "Terminated Date")]
        public DateTime? TerminatedDate { get; set; }
    }
}

File: employee-management-app/src/app/models/employee.ts
export interface IEmployee {
  birthDate: Date;
  employeeId: number;
  firstName: string;
  lastName: string;
  personId: number;
  employeeNum: string;
  employedDate: Date;
  terminatedDate?: Date;
}

export class Employee implements IEmployee {
  employeeId: number = 0;
  birthDate: Date = new Date();
  firstName: string = '';
  lastName: string = '';
  personId: number = 0;
  employeeNum: string = '';
  employedDate: Date = new Date();
  terminatedDate?: Date | undefined;

  constructor(
    firstName: string,
    lastName: string,
    personId: number,
    employeeNum: string,
    employedDate: Date,
    terminatedDate?: Date) {

  }


}